src/app/server/Mdns.cpp

```diff
@@ -115,6 +116,12 @@ CHIP_ERROR AdvertiseCommisioning()
 
     auto advertiseParameters = chip::Mdns::CommissionAdvertisingParameters().SetPort(CHIP_PORT).EnableIpV4(true);
 
+    // TODO: this code assumes the device always boots into commissioning mode
+    // fix this once we are able to configure default boot behavior
+    // and when we can determine whether we have been commissioned yet
+    bool notYetCommissioned = true;
+    advertiseParameters.SetCommissioningMode(notYetCommissioned, false);
+
```

Comment:
This seems like a big TODO :) Can we take a stab at this?

Reply:
I fully plan to but trying not to go too far in one PR